app/views/todo_items/_form.html.erb

```diff
@@ -1,21 +1,19 @@
 <% if @todo_item.errors.any? %>
-<div id="error_explanation">
-  <h2><%= pluralize(@todo_item.errors.count, "error") %> prohibited this todo item from being saved:</h2>
-
-  <ul>
-    <% @todo_item.errors.full_messages.each do |msg| %>
-      <li><%= msg %></li>
-    <% end %>
-  </ul>
-</div>
+  <div id="error_explanation">
+    <h2><%= pluralize(@todo_item.errors.count, "error") %> prohibited this todo item from being saved:</h2>
+    <ul>
+      <% @todo_item.errors.full_messages.each do |msg| %>
+        <li><%= msg %></li>
+      <% end %>
+    </ul>
+  </div>
 <% end %>
 
-
-<%= form.label :content %>
-<%= form.text_field :content %>
-<%= form.label :deadline %>
-<%= form.datetime_select :deadline, :default => 3.days.from_now %>
-<%= form.label :tag_list %>
-<%= form.text_field :tag_list %>
-
-<%= form.submit "Save", id: "save-todo-item"  %>
+<% present(@todo_item, TodoItemPresenter) do |todo_item| %>
```

Comment:
Opakowywanie obiektu w presenter powinno się odbywać na poziomie kontrolera - w ten sposób niezależnie od tego w którym miejscu widoku do niego się będziesz dobierać to te same metody będą działały. W tej sytuacji W linijce 11 `@todo_item.tag_list` rzuci błędem, a w 13 będzie poprawne - to prowadzi do bardzo trudnych do znalezienia błędów.

Tak więc w kontrolerze robisz:

```ruby
@todo_item = TodoItemPresenter.new(@todo_item)
```